purchase_service_picking/models/service_picking.py

```diff
@@ -0,0 +1,447 @@
+        return {
+            "name": _("Immediate Transfer?"),
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": "service.immediate.transfer",
+            "views": [(view.id, "form")],
+            "view_id": view.id,
+            "target": "new",
+            "context": dict(
+                self.env.context, default_picking_ids=[(4, p.id) for p in self]
+            ),
+        }
+
+    def _action_generate_backorder_wizard(self):
+        view = self.env.ref("purchase_service_picking.view_backorder_confirmation")
+        return {
+            "name": _("Create Backorder?"),
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": "service.backorder.confirmation",
+            "views": [(view.id, "form")],
+            "view_id": view.id,
+            "target": "new",
+            "context": dict(
+                self.env.context, default_picking_ids=[(4, p.id) for p in self]
+            ),
+        }
+
+
+class ServicePickingLine(models.Model):
+    _name = "service.picking.line"
+    _description = "Service Picking Line"
+    _order = "sequence, id"
+
+    picking_id = fields.Many2one(
+        comodel_name="service.picking",
+        string="Picking",
+        index=True,
+        required=True,
+        ondelete="cascade",
+    )
+    sequence = fields.Integer(string="Sequence", default=10)
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        default=lambda self: self.env.company,
+        index=True,
+        required=True,
+    )
+    purchase_line_id = fields.Many2one(comodel_name="purchase.order.line")
+    state = fields.Selection(related="picking_id.state")
+    name = fields.Char(string="Description", index=True, required=True)
+    origin_returned_line_id = fields.Many2one(
+        comodel_name="service.picking.line",
+        string="Origin return line",
+        copy=False,
+        index=True,
+        help="Line that created the return line",
+    )
+    returned_line_ids = fields.One2many(
+        comodel_name="service.picking.line",
+        inverse_name="origin_returned_line_id",
+        string="All returned lines",
+    )
+    date = fields.Datetime(
+        string="Date Scheduled",
+        default=fields.Datetime.now,
+        index=True,
+        required=True,
+    )
+    date_deadline = fields.Datetime(
+        string="Deadline",
+        readonly=True,
+        help="Date Promise to the customer on the top level document (SO/PO)",
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Product",
+        check_company=True,
+        domain="""[('type', '=', 'service'), '|', ('company_id', '=', False),
+        ('company_id', '=', company_id)]""",
+        index=True,
+        required=True,
+        states={"done": [("readonly", True)]},
+    )
+    product_qty = fields.Float(
```

Comment:
Use latest version field nomenclature.

Reply:
I think you mean the proper name is `product_uom_qty` (_Demand_), I change it.